rust/src/nlp/src/npl.rs

```diff
@@ -0,0 +1,27 @@
+
+use rust_bert::pipelines::common::ModelType;
+use rust_bert::pipelines::question_answering::{QuestionAnsweringModel, QaInput};
+use std::error::Error;
+
+// process user input and extract relevant information
+pub(crate) fn process_input(input_text: &str) -> Result<String, Box<dyn Error>> {
+    //  QuestionAnsweringModel
+    let qa_model = QuestionAnsweringModel::new(ModelType::Bert);
```

Comment:
Are you actually running this in your console? It seems to me that `QuestionAnsweringModel::new()` should take more parameters. 

```
error[E0061]: this function takes 4 arguments but 1 argument was supplied
   --> src/nlp/src/main.rs:11:20
    |
11  |     let qa_model = QuestionAnsweringModel::new(ModelType::Bert);    
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^----------------- three arguments of type `&Path`, `&Path`, and `tch::wrappers::device::Device` are missing
    |
note: associated function defined here
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rust-bert-0.5.3/src/pipelines/question_answering.rs:202:12
    |
202 |     pub fn new(vocab_path: &Path, config_path: &Path, weights_path: &Path, device: Device)
    |            ^^^
help: provide the arguments
    |
11  |     let qa_model = QuestionAnsweringModel::new(ModelType::Bert, /* &Path */, /* &Path */, /* tch::wrappers::device::Device */); 
```

Reply:
yes your right, still figuring out the best workflow for this.